src/net40/Radical.Windows/Presentation/Behaviors/DataGridBehaviors/ScrollIntoViewForDataGrid.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Windows.Controls;
+using Topics.Radical.Windows.Behaviors;
+
+namespace Topics.Radical.Windows.Presentation.Behaviors
+{
+    class ScrollIntoViewForDataGrid : RadicalBehavior<DataGrid>
+    {
+        protected override void OnAttached()
+        {
+            base.OnAttached();
+            this.AssociatedObject.SelectionChanged += AssociatedObject_SelectionChanged;
+        }
+
+        void AssociatedObject_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            if (sender is DataGrid)
+            {
+                DataGrid dataGrid = (sender as DataGrid);
```

Comment:
can be a direct cast:
```
var dataGrid = (DataGrid)sender;
```

Or you can also use Radical `As<>` extension:

```
sender.As<DataGrid>( dataGrid => {
   //this will invoked only if the cast succeeds
});
```